protarrow/proto_to_arrow.py

```diff
@@ -67,6 +67,12 @@
 }
 
 
+class ProtarrowCycleError(Exception):
```

Comment:
Typically the library throws generic `TypeError`. You'd want to specialize error if there's a way to recover from it at runtime. For example here, the user could turn `purge_cyclical_messages` on and retry in the `except` statement. But it feels like a convoluted use case. 

I'd suggest just using `TypeError`, or inherit from `TypeError` here.